packages/flutter/lib/src/material/data_table.dart

```diff
@@ -174,6 +188,11 @@ class DataRow {
   /// table.
   final List<DataCell> cells;
 
+  /// {@macro flutter.material.dataColumn.heading}
+  ///
+  /// Summarizes the information displayed in this [DataRow].
+  final String? heading;
```

Comment:
since this is a summary for the entire row, consider maybe call this semanticsSummary